Azure/sonic-snmpagent
# [RFC4292][Namespace]: Fix implementation of RouteUpdater for multi-asic platform

For multi-asic platforms, the default route was getting retrieved from only
the last namespace as the result dictionary was being written with the same key.
To avoid this, modify implementation to retrieve default routes from all
front end asic namespaces. Update implementation to ensure that only
external routes are displayed in the result.

**- What I did**
- Fix RouteUpdater implementation and add unit-test to test multiple namespaces.

**- How I did it**
- Fix RouteUpdater implementation to retrieve default routes from all front end asic namespaces.
- Fix implementation to ensure only external routes are displayed in the result.
- Add test case to test RFC4292 implementation for multiple namespaces scenario.

**- How to verify it**
- Unit-test cases should pass.
- Update snmp docker with the changes on single asic and multi-asic platform.
- On single asic platform SNMP walk of 1.3.6.1.2.1.4.24.4 should not change before and after the fix.
- On multi-asic platform, SNMP results should show external default route information.

**- Description for the changelog**
